README.md
sliders.js
==========

sliders.js is a tiny jQuery slideshow plugin.

##1. Setup

Include the latest jQuery Library as well as sliders.js just before your closing body tag

```html
<script src="http://code.jquery.com/jquery-1.10.2.min.js"></script>
<script src="/js/sliders.min.js"></script>
<script src="/js/sliders.min.css"></script>
```

##2. HTML Markup
*note: the .page, .pager and data-target values are important
```html
<div class="=sliders"> <!--this is the default class in sliders.css-->
  <figure class="slides">
   <img src='/img/sliders.png' />
  <figcaption> <!--optional-->
   Your captions
  </figcaption>
  </figure>
  <div class="page" id="arrow_right" data-target="next"></div> <!--uses data-target to move through slides-->
  <div class="page" id="arrow_left" data-target="prev"></div>
  <ul class="pager"></ul> <!--optional but necessary for pager bullets-->

</div>
```
##3. CSS
```css
.sliders {
position: relative;
float: left;
width: 100%;
height: 400px;
}

.sliders > figure {
position: absolute;
opacity: 0;
width: 100%;
top: 0;
left: 0;
right: 0;
bottom: 0;
height: 100%;
}

.sliders > figure img {
width: 100%;
height: 100%;
z-index: 1;
}
```

##4. Usage
Start the show....

```javascript
$(".sliders").sliders();
```
or with options:
```javascript
$('.slideshow').sliders({
  "auto" : false,             //default false, auto start
  "speed" : 500,              //speed of the transition (default 500)
  "timeout" : 7000,           //time between slides (default 7000)
  "bullets" : false,          //pager bullets (default false)
  "animate": false,           //animates the slide, use in conjuction with direction
  "direction": "left"         //direction of the slide animation (must set this when using animate:true)
  easing: 'jswing'            //option easing functions (ease In only, see http://easings.net/ for your options)
});
```

##5. Demo
For a more in depth view, see the example
http://www.dylan-foster.com/sliders

##Credits
[jQuery](http://api.jquery.com/)<br>

##Contact
Copyright (C) 2014 Dylan foster<br>
[Fostered Development](http://www.dylan-foster.com)<br>
[Github](https://github.com/dylan947/)<br>
[@dylfos](http://twitter.com/dylfos)<br>
[+Fostered Development](https://plus.google.com/b/103850011544407258916/103850011544407258916/about)





History.md

2.0.5 / 2014-10-26
==================

 * [BUG FIX] - Enhances pagination dot detection

2.0.3 / 2014-10-25
==================

 * [BUG FIX] - only checks for pagination when existent


gulpfile.js
"use strict";

var gulp = require("gulp");
var eslint = require("gulp-eslint");
var jsonlint = require("gulp-jsonlint");
var uglify = require("gulp-uglify");
var minifyCSS = require("gulp-minify-css");
var rename = require("gulp-rename");

var css = ["./src/**/*.css"];
var js = ["./src/**/*.js"];

gulp.task("default", ["lint", "build", "copy"]);

gulp.task("build", function(){
  gulp.src(css)
  .pipe(minifyCSS())
  .pipe(rename(function(file){
    file.basename += ".min";
    file.ext = ".css";
  }))
  .pipe(gulp.dest("./build"));

  gulp.src(js)
  .pipe(uglify({ mangle: false }))
  .pipe(rename(function(file){
    file.basename += ".min";
    file.ext = ".js";
  }))
  .pipe(gulp.dest("./build"));
});

gulp.task("copy", function(){
  gulp.src("./src/img/*", { base: "./src/img"})
  .pipe(gulp.dest("./build/img"));
});

gulp.task("lint", function(){
  gulp.src([].concat(js, "./gulpfile.js"))
  .pipe(eslint())
  .pipe(eslint.format("stylish", console.error));

  gulp.src(["./package.json", "./sliders.js.jquery.json"])
  .pipe(jsonlint())
  .pipe(jsonlint.reporter("stylish", console.error));
});


src/sliders.js
"use strict";

(function ($) {

  var methods = {
    init: function (options, easing) {
      return this.each(function () {
        var settings = {
          "auto": false,
          "speed": 500,
          "timeout": 7000,
          "bullets": false,
          "animate": false,
          "direction": [
            "left", "right", "top", "bottom"
          ],
          easing: "jswing"
        };

        if (options) {
          $.extend(settings, options, easing);

        }

        this.settings = settings;
        var $page, $pager;
        var $cont = $(this);
        var img = $cont.children("figure");
        var count = img.length;
        var fading = false;
        var slideTimer;
        var activeSlide;
        var newSlide;

        if (settings.bullets) {
          $pager = $cont.find(".pager");

          for (var i = 0; i < count; i++) {
            $pager.append("<li class=\"page\" id=" + i + ">" + i + "</li>");
            $("li#" + i).attr("data-target", i);
          }

          $page = $pager.children(".page");
          $page.eq(0).addClass("active");

        }

        function waitForNext() {
          slideTimer = setTimeout(function () {
            changeSlides("next");
          }, settings.timeout);
        }

        function animateSlides(activeNdx, newNdx) {
          function cleanUp() {
            img.eq(activeNdx).removeAttr("style");
            activeSlide = newNdx;
            fading = false;
            waitForNext();
          }

          if (fading || activeNdx === newNdx) {
            return false;
          }

          fading = true;

          if ($pager.length) {
            $page.removeClass("active").eq(newSlide).addClass("active");
          }

          img.eq(activeNdx).css("z-index", 3);
          img.eq(newNdx).css({
            "z-index": 2,
            "opacity": 1
          });


          if (settings.animate) {
            switch (settings.direction) {
            case "left":
              img.eq(activeNdx).stop().animate({
                  "left": "-9999px"
                }, settings.speed, settings.easing,
                function () {
                  cleanUp();
                });
              break;

            case "right":
              img.eq(activeNdx).stop().animate({
                  "right": "-9999px",
                  "left": "9999px"
                }, settings.speed, settings.easing,
                function () {
                  cleanUp();
                });
              break;

            case "top":
              img.eq(activeNdx).stop().animate({
                  "top": "-9999px",
                  "bottom": "9999px"
                }, settings.speed, settings.easing,
                function () {
                  cleanUp();
                });
              break;

            case "bottom":
              img.eq(activeNdx).stop().animate({
                  "top": "9999px",
                  "bottom": "-9999px"
                }, settings.speed, settings.easing,
                function () {
                  cleanUp();
                });
              break;

            }
          } else {

            img.eq(activeNdx).stop().animate({
                "opacity": 0
              }, settings.speed,
              function () {
                cleanUp();
              });
          }


        }

        function changeSlides(target) {
          if (target === "next") {
            newSlide = activeSlide + 1;
            if (newSlide > count - 1) {
              newSlide = 0;
            }
          } else if (target === "prev") {
            newSlide = activeSlide - 1;
            if (newSlide < 0) {
              newSlide = count - 1;
            }
          } else {
            newSlide = target;
          }
          animateSlides(activeSlide, newSlide);
        }

        img.eq(0).css({
          "opacity": 1
        });

        activeSlide = 0;

        if (settings.auto) {
          waitForNext();
        }

        $cont.find(".page").bind("click", function () {
          var target = $(this).attr("data-target");
          clearTimeout(slideTimer);
          changeSlides(target);
        });
      });
    }
  };

  $.fn.sliders = function (settings) {
    return methods.init.apply(this, arguments);
  };
})(jQuery);
